### PR TITLE
docs: add AR-04 for backend-contract invariant enforcement

### PR DESCRIPTION
**Pull request by timurgilfanov**
## Summary
Extracts the backend-contract invariant documentation into a dedicated Architecture Rule. Split out of #388 (chat paging refresh) as it is out of scope for that PR.

> [!NOTE]
> Stacked on `issue/149` because the documented code (`InvalidParticipantContractException` in `LocalSyncDataSourceImpl`) is introduced there and not yet on `main`. Retarget this PR to `main` once #388 merges.

## Changes
- Add `docs/architecture/AR-04-backend-contract-invariants.md`: validate backend-contract invariants once at the persistence boundary inside the write transaction, surface violations as handled data-source errors (rolled back), and trust the invariant downstream without defensive guards. A post-boundary lookup failure crashes because it signals an internal bug. Uses the current-user participant contract as the worked example.
- Slim the CLAUDE.md "Backend Chat Contract" bullet to a one-line pointer at AR-04, and correct the earlier wording: a violating backend payload is a handled error, not a crash.

## Important Design Decision
The rule generalizes the producer/consumer split already present in the code: enforcement is centralized at the single persistence chokepoint (`LocalSyncDataSourceImpl.validateParticipantContract`, inside `database.withTransaction`) so every downstream reader inherits the guarantee and uses total lookups (`first {}`), never `firstOrNull` guards. The in-transaction throw is deliberate — it triggers atomic rollback, which a `ResultWithError` returned inline would not.

## Testing
Documentation only — no code behavior changes.
- [ ] Unit tests added or updated
- [ ] Manually tested
- [x] Edge cases considered

## Screenshots / Demo
N/A — documentation only.

## Breaking Changes
None.

## Follow-ups
- [x] Retarget this PR to `main` after #388 merges.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: app/src/main/res/values-de/strings.xml [hunk 1]
```diff
@@ -49,6 +49,11 @@
     <string name="settings_language_german">Deutsch</string>
     <string name="settings_language_change_failed">Sprache kann nicht geändert werden</string>
 
+    <!-- Chat Screen -->
+    <string name="chat_messages_failed_to_load">Nachrichten konnten nicht geladen werden</string>
+    <string name="chat_paging_retry">Erneut versuchen</string>
+    <string name="chat_update_failed">Chat konnte nicht aktualisiert werden</string>
+
```
Comment (timurgilfanov) [inline app/src/main/res/values-de/strings.xml]:
```suggestion
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -49,6 +49,11 @@
     <string name="settings_language_german">German</string>
     <string name="settings_language_change_failed">Cannot change language</string>
 
+    <!-- Chat Screen -->
+    <string name="chat_messages_failed_to_load">Messages failed to load</string>
+    <string name="chat_paging_retry">Retry</string>
+    <string name="chat_update_failed">Could not update chat</string>
+
```
Comment (timurgilfanov) [inline app/src/main/res/values/strings.xml]:
```suggestion
```
